app/uk/gov/hmrc/transitmovementsvalidator/services/XmlValidationService.scala

```diff
@@ -76,38 +78,66 @@ class XmlValidationServiceImpl @Inject() (implicit ec: ExecutionContext) extends
             saxParser =>
               Using(source.runWith(StreamConverters.asInputStream(20.seconds))) {
                 xmlInput =>
-                  val parser = saxParser.newSAXParser.getParser
-
+                  val inputSource = new InputSource(xmlInput)
                   val errorBuffer: mutable.ListBuffer[XmlSchemaValidationError] =
                     new mutable.ListBuffer[XmlSchemaValidationError]
-
-                  parser.setErrorHandler(new ErrorHandler {
-                    override def warning(error: SAXParseException): Unit = {}
-
-                    override def error(error: SAXParseException): Unit =
-                      errorBuffer += XmlSchemaValidationError.fromSaxParseException(error)
-
-                    override def fatalError(error: SAXParseException): Unit =
-                      errorBuffer += XmlSchemaValidationError.fromSaxParseException(error)
-                  })
-
-                  val inputSource = new InputSource(xmlInput)
+                  var elementValue = ""
+                  var rootTag      = ""
+
+                  val parser = saxParser.newSAXParser.parse(
+                    inputSource,
+                    new DefaultHandler {
+                      var inMessageTypeElement = false
+
+                      override def characters(ch: Array[Char], start: Int, length: Int): Unit =
+                        if (inMessageTypeElement) {
+                          elementValue = new String(ch, start, length)
+                        }
+
+                      override def startElement(uri: String, localName: String, qName: String, attributes: Attributes) = {
+                        if (qName.startsWith("ncts:")) {
```

Comment:
You cannot assume that the prefix will be this. Traders can use whatever prefix they like, as long as the prefix and namespace match up.

An assumption like this actually bit me when I was setting up connectivity on QA, ERMIS didn't use the prefixes the EIS spec gave. Still totally valid, but the hardcoded prefixes broke things.

Reply:
(that does remind me, I should check this assumption for trader -> ERMIS messages in the router)

Reply:
yes, updated to override startPrefixMapping to get the namespace